packages/sit-onyx/src/components/OnyxSlider/OnyxSlider.vue

```diff
@@ -0,0 +1,725 @@
+const skeleton = useSkeletonContext(props);
+
+const handleChange = (values: number[]) => {
+  emit("update:modelValue", values);
+};
+
+const handleCommit = (values: number[]) => {
+  emit("update:committed", values);
+};
+
+const handleDecreaseByIcon = () => {
+  if (disabled.value) return;
+
+  if (modelValue.value.length === 1) {
+    const currentValue = modelValue.value[0] ?? props.min;
+    const stepValue = props.shiftStep ?? props.step ?? 1;
+    const newValue = Math.max(currentValue - stepValue, props.min);
+    const newValues = [newValue];
+    handleChange(newValues);
+    handleCommit(newValues);
+  }
+};
+
+const handleIncreaseByIcon = () => {
+  if (disabled.value) return;
+
+  if (modelValue.value.length === 1) {
+    const currentValue = modelValue.value[0] ?? props.min;
+    const stepValue = props.shiftStep ?? props.step ?? 1;
+    const newValue = Math.min(currentValue + stepValue, props.max);
+    const newValues = [newValue];
+    handleChange(newValues);
+    handleCommit(newValues);
+  }
+};
+
+const { min, max, step, marks, orientation } = toRefs(props);
+
+const {
+  elements: { root, rail, track, thumbContainer, thumbInput, mark, markLabel },
+  state: { focusedThumbIndex, activeThumbIndex, isDragging, marksList },
+  internals: { isMarkActive, valueToPercent, axis },
+} = createSlider({
+  values: modelValue,
+  min,
+  max,
+  step,
+  marks,
+  orientation,
+  onChange: handleChange,
+  onCommit: handleCommit,
+});
+
+const isValueControl = computed(() => props.control === "value");
+/**
+ * Icon control works only when there is a single thumb.
+ */
+const isIconControl = computed(() => props.control === "icon" && modelValue.value.length === 1);
+</script>
+
+<template>
+  <div
+    v-if="skeleton"
+    v-bind="rootAttrs"
+    :class="['onyx-component', 'onyx-slider-skeleton', densityClass]"
+  >
+    <div
+      class="onyx-slider-skeleton__container"
+      :class="{
+        'onyx-slider-skeleton__container--vertical': props.orientation === 'vertical',
+      }"
+    >
+      <OnyxSkeleton v-if="isValueControl" class="onyx-slider-skeleton__value-control" />
+      <OnyxSkeleton v-if="isIconControl" class="onyx-slider-skeleton__icon-control" />
+      <div class="onyx-slider-skeleton__root">
+        <OnyxSkeleton class="onyx-slider-skeleton__track" />
+      </div>
+      <OnyxSkeleton v-if="isValueControl" class="onyx-slider-skeleton__value-control" />
+      <OnyxSkeleton v-if="isIconControl" class="onyx-slider-skeleton__icon-control" />
+    </div>
+  </div>
+
+  <div
+    v-else
+    v-bind="rootAttrs"
+    class="onyx-component onyx-slider"
+    :class="[
+      {
+        'onyx-slider--vertical': props.orientation === 'vertical',
+        'onyx-slider--inverted': props.trackMode === 'inverted',
+        'onyx-slider--disabled': disabled,
+        'onyx-slider--is-dragging': isDragging,
+        'onyx-slider--is-focused': focusedThumbIndex !== -1,
+        'onyx-slider--is-active': activeThumbIndex !== -1,
```

Comment:
Are those 3 classes needed? Can't we use native CSS selectors like `:focus-within` or `:active`?